core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonProtocolUnmarshaller.java

```diff
@@ -224,23 +217,111 @@ public T unmarshall(JsonUnmarshallerContext context,
 
     public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
                                                     SdkHttpFullResponse response) throws IOException {
-        JsonNode jsonNode = hasJsonPayload(sdkPojo, response) ? parser.parse(response.content().get()) : null;
-        return unmarshall(sdkPojo, response, jsonNode);
+        if (!hasJsonPayload(sdkPojo, response)) {
+            return unmarshallResponse(sdkPojo, response);
+        }
+        if (hasExplicitJsonPayloadMember(sdkPojo)) {
+            return unmarshallResponse(sdkPojo, response);
+        }
+        if (hasMixedLocations(sdkPojo)) {
+            unmarshallFromJson(sdkPojo, response.content().get());
+            return unmarshallResponse(sdkPojo, response);
+        }
+        return unmarshallFromJson(sdkPojo, response.content().get());
```

Comment:
Are these `get()`s guaranteed to be safe here?

Reply:
At this point yes, otherwise `hasJsonPayload` above returns false and the first branch will be used.